R/find_exports.R

```diff
@@ -2,18 +2,22 @@ find_exports <- function(clean_lns) {
   ids <- find_extendr_attrs_ids(clean_lns)
   start <- ids
   end <- dplyr::lead(ids, default = length(clean_lns) + 1L) - 1L
-  purrr::map2_dfr(
-    start,
-    end,
-    ~ extract_meta(clean_lns[seq(.x, .y, by = 1L)])
-  ) %>%
-    # Keeps only name, type (fn|impl) and lifetime of impl
-    # if present.
-    dplyr::transmute(
-      .data$name,
-      type = dplyr::if_else(is.na(.data$impl), "fn", "impl"),
-      .data$lifetime
-    )
+
+  # instantiate list to fill
+  res <- rlang::new_list(length(start))
+
+  for (i in seq_along(start)) {
+    res[[i]] <- extract_meta(clean_lns[start[i]:end[i]])
+  }
+
+  # bind results together into tibble
+  res <- do.call(dplyr::bind_rows, res)
+
+  # Keeps only name, type (fn|impl) and lifetime of impl if present.
+  res[["type"]] <- dplyr::coalesce(res[["impl"]], res[["fn"]])
+
+  res[c("name", "type", "lifetime")]
+
```

Comment:
Any reason rewriting to a `for` loop rather than a `map` function?

Reply:
A few reasons. Primarily is that when an error is encounterred in `map2_dfr()` the error message that is propagated is from purrr first thus failing the testthat expectations.  Note the "in index: 1." message. 

```
Error in `map2()`:
ℹ In index: 1.
Caused by error in `extract_meta()`:
! Rust code contains invalid attribute macros.
✖ No valid `fn` or `impl` block found in the following sample:
#[extendr]
let invalid_var = ();
```

By using a for loop, we avoid this. 

Secondly, the for loop is faster and, to my eyes, more readable than the anonymous function piped into a transmute statement using a data mask.

Reply:
If you're fine with purrr, then some test expectations need to be altered so that they expect the purrr message not the extendr one.

Reply:
I get the `purrr` argument, but I am not convinced we need to use a plain `for` loop.

Reply:
Is the decision then to use purrr and change the test exectation?